proto/composerapis/astria/auctioneer/v1alpha1/bundle.proto

```diff
@@ -0,0 +1,39 @@
+syntax = "proto3";
+
+package astria.auctioneer.v1alpha1;
+
+import "google/protobuf/timestamp.proto";
+import "google/protobuf/empty.proto";
+import "astria/execution/v1alpha2/execution.proto";
+import "astria/auctioneer/v1alpha1/optimistic_block.proto";
+
+// Information for the bundle submitter to know how to submit the bundle.
+// The fee and base_sequencer_block_hash are not necessarily strictly necessary
+// it allows for the case where the server doesn't always send the highest fee
+// bundles after the previous but could just stream any confirmed bundles.
+message Bundle {
+    // The fee that can be expected to be received for submitting this bundle.
+    // This allows the bundle producer to stream any confirmed bundles they would be ok
+    // with submitting. Used to avoid race conditions in received bundle packets. Could
+    // also be used by a bundle submitter to allow multiple entities to submit bundles.
+    uint64 fee = 1;
+    // The byte list of transactions to be included.
+    repeated bytes transactions = 2;
+    // The base_sequencer_block_hash is the hash from the base block this bundle
+    // is based on. This is used to verify that the bundle is based on the correct
+    // Sequencer block.
+    bytes base_sequencer_block_hash = 3;
+    // The hash of previous rollup block, on top of which the bundle will be executed as ToB.
+    bytes prev_rollup_block_hash = 4;
+}
+
+// Geth will serve this to the Auctioneer
+service BundleService {
+    // Stream blocks from the Auctioneer to Geth for optimistic execution. Geth will stream back
+    // metadata from the executed blocks.
+    rpc ExecuteOptimisticBlocks(stream BaseBlock) returns (stream astria.execution.v1alpha2.Block);
```

Comment:
```suggestion
    rpc ExecuteOptimisticBlock(stream BaseBlock) returns (stream astria.execution.v1alpha2.Block);
```

however i'm not sure why this is needed when we already have an `ExecuteBlock` method - they seem to be taking the same inputs and doing the same thing (just executing a block and returning it)?

Reply:
Its not just doing the same thing as conductor. When geth receives blocks from `ExecuteOptimisticBlock`, it does the following:
1. Execute the block and update the optimistically synced fork
2. Update the legacy mempool to use the optimistic block to perform stateful checks on
3. Clear the mempool of transactions to allow searchers to send transactions valid on a new block.

(2) and (3) will be done with events

It is also a good idea to separate it out from ExecuteBlock for the following reasons:
1. We are using a stream here instead of a unary RPC for efficiency 
2. ExecuteBlock holds a lock when executing blocks. it might be better to separate out the calls and let only conductor only interact with ExecuteBlock to avoid any unnecessary contentions. 
3. Conductor is already an audited path, changes in conductor would involve us to re-audit it.

Reply:
>  the optimistically synced fork

does an auctioneer-geth have two separate forks then? an optimistic fork and a fork from conductor blocks? i'm not sure this is functionally different from the existing head-soft-firm block structure in geth, optimistic can be head, soft and firm as the same as now

does `ExecuteOptimisticBlock` atomically update the block to be head as well then? if so, it'll need a lock, as `SetCommitmentState` also updates the head of chain

Reply:
yes so currently, we are maintaining 2 separate forks. An optimistic fork and a fork from conductor blocks. We are not updating the block to be head in geth as per the current impl. 
We maintain a separate block pointer for the optimistic block as per: https://github.com/astriaorg/flame/blob/bharath/add-mempool-clearing-event/core/blockchain.go#L242
This is updated in `ExecuteOptimisticBlock` method as per https://github.com/astriaorg/flame/blob/bharath/add-mempool-clearing-event/grpc/execution/server.go#L302 (I haven't hooked it up with the grpc streams interface yet, so this name will change)
Essentially, we do the following:
1. We insert the optimistic block into the chain without setting the head
2. We set the optimistic block pointer to this block header to reference the state in the future

Reply:
my thinking is that, by this method we don't need to hold the lock held in `ExecuteBlock` and `UpdateStateCommitment` because the pieces of data being operated on are different and do not interfere.